### PR TITLE
Domains: Mapping and Transfer pages standardization, take two

### PR DESCRIPTION
**Pull request by sixhours**
* Change pricing display on Transfers page to match placement from Mapping page
* Disable "Add" button on Mapping page until a proper domain name is input, a la Transfers page
* Moves Transfer fine print/Learn More underneath the form to put the focus on the form/pricing.

Is it OK to move that explanatory text from the Transfer page underneath? Does the pricing stand out enough? What do y'all think?

**Before this PR**

screen shot 2018-07-26 at 11 32 50 am

screen shot 2018-07-26 at 11 33 26 am

**After this PR**

Lots of visual changes here, so I've added everything to a gallery: https://cloudup.com/c0giKp1KWjo

**Steps to test**

* Switch to this PR and navigate to `/start/domains/use-your-domain/`
* Click "Transfer to WordPress.com" to note design changes; make sure the form still works.
* Click "Buy Domain Mapping" to note design changes; make sure the form still works.

### CODE REVIEW

File: client/components/domains/transfer-domain-step/index.jsx [hunk 1]
```diff
@@ -179,26 +179,10 @@ class TransferDomainStep extends React.Component {
...
-												target="_blank"
-											/>
-										),
-									},
-								}
-							) }
-						</div>
 					</div>
 
+					<div className="domain-product-price">{ domainProductPrice }</div>
```
Comment (jancavan) [inline client/components/domains/transfer-domain-step/index.jsx]: I think this causing tests to fail. We should rename class: `transfer-domain-step__ prefix`

Comment (jancavan) [reply client/components/domains/transfer-domain-step/index.jsx]: screen shot 2018-07-26 at 10 00 10 am

Comment (danhauk) [reply client/components/domains/transfer-domain-step/index.jsx]: I think it's just looking for `transfer-domain-step__` as a prefix, not necessarily needing `prefix` in the class name. Something like `transfer-domain-step__price` would make sense I think.

Comment (sixhours) [reply client/components/domains/transfer-domain-step/index.jsx]: Aha, good point. I was trying to reuse the same class name between the two sections for consistency, but one uses a Component and the other does not. Will change that!

Comment (jancavan) [reply client/components/domains/transfer-domain-step/index.jsx]: Yeah, sorry, I meant to say the class needs to be prefixed with  transfer-domain-step__` :)

**Comment by jancavan** [comment]
I think "Included in paid plans" should have the same font size/weight in both pages?

Domain Transfer:
screen shot 2018-07-26 at 10 46 57 am

Domain Mapping:
screen shot 2018-07-26 at 10 47 12 am

Also, do you think "included in paid plans" should be green?

**Comment by sixhours** [comment]
@jancavan Ahh yes, good catch. I used a function to add a class to change the display depending on the output; again, a weird side effect of one page using a component and the other not. We'll want a dev to look into it to make sure I did this right. :)

**Comment by sixhours** [comment]
Hmm, not sure why those Canary tests are failing.

**Comment by jancavan** [comment]
"Included in paid plans" color LGTM!

Just noticed this now, but do you think about moving it to the left when it hits `<960px` instead of `<660px` so this doesn't happen?

(This is around 850ish):
screen shot 2018-07-26 at 1 50 14 pm

I'm unable to get this section to display the price (ie $13 /month), but that'll look the same, correct?

**Comment by sixhours** [comment]
> I'm unable to get this section to display the price (ie $13 /month), but that'll look the same, correct?

It will look like this: 

No green, since we haven't done that in previous domains screens. What I do wonder about is whether the domain transfers page is a /year cost or a one-time cost. Maybe someone from Cobalt can help clarify that.

**Comment by jancavan** [comment]
> No green, since we haven't done that in previous domains screens. 

Right. Sorry, I meant to ask if its current look won't change :) Looks like it'll stay the same, so LGTM.

File: client/components/domains/transfer-domain-step/index.jsx [hunk 2]
```diff
@@ -164,11 +164,31 @@ class TransferDomainStep extends React.Component {
 		return domainProductPrice;
 	};
 
+	getProductPriceClass = () => {
+		const { cart, domainsWithPlansOnly, isSignupStep, selectedSite } = this.props;
+		const { searchQuery } = this.state;
+		const domainsWithPlansOnlyButNoPlan =
```
Comment (jsnmoon) [inline client/components/domains/transfer-domain-step/index.jsx]: What does this mean? That the domain is offered only if bundled with a plan?

**Comment by sixhours** [comment]
> When I navigate back in /start/domains/transfer/ or /start/domains/mapping/ using the Back button at the bottom of the page, I should be brought back to /start/domains/use-your-domain/. Currently, it brings me back to /start/domains/ instead.

Yeah, the Back buttons throughout signup produce some interesting and unexpected results. I'm going to open a separate ticket for this.

> The mapping copy for /start/domains/use-your-domain/ says $13.00 per year plus registration costs at your current provider while the transfer copy shows Included in paid plans. We should aim to make this more consistent since mapping is included in the plan pricing.

Oh, good catch. @jancavan what do you think? I think the green text adds a bit of visual interest, but I do think we should be consistent. Using green on the search page is a bit overpowering:

screen shot 2018-07-30 at 8 58 03 pm

> /start/domains/mapping/ has very little copy describing what mapping actually is. We have a lot of copy from /start/domains/use-your-domain/ we could probably reuse here.

Agreed, I think we could add some similar "fine print", like what the Transfers page has, with a link to learn more.

File: client/components/domains/transfer-domain-step/index.jsx (hunk 2 above)
Comment (sixhours) [reply client/components/domains/transfer-domain-step/index.jsx]: Yes, I modeled this after the `getProductPriceText()` function. If I'm reading it right, this returns true when a domain needs to be bundled with a plan but currently no plan is selected/active on the given site, or when we're in signup.

**Comment by jancavan** [comment]
> The mapping copy for /start/domains/use-your-domain/ says $13.00 per year plus registration costs at your current provider while the transfer copy shows Included in paid plans.

> /start/domains/mapping/ has very little copy describing what mapping actually is. We have a lot of copy from /start/domains/use-your-domain/ we could probably reuse here.

+1 to the two above. We should address those in a separate PR.

> I think the green text adds a bit of visual interest, but I do think we should be consistent. Using green on the search page is a bit overpowering.

Yeah, it does.. and I'm worried that highlighting "Free" might result in more people selecting this plan. Maybe it's best to just keep it gray? What do you think?

**Comment by sixhours** [comment]
@jancavan Thanks for the second look, I think the gray works fine in this case.

**Comment by jancavan** [comment]
Sorry we had to change it back. It's still green on the Domain Mapping page? Other than that, LGTM  👍

File: client/components/domains/transfer-domain-step/index.jsx (hunk 2 above)
Comment (jsnmoon) [reply client/components/domains/transfer-domain-step/index.jsx]: 👍

**Comment by sixhours** [comment]
Oops, forgot about that. :) Green text now gray on the Domain Mapping page as well.

**Pull request merged**